actors/miner/src/state.rs

```diff
@@ -877,46 +858,42 @@ impl State {
                 amount_unlocked
             ));
         }
-        // add locked funds now
-        vesting_funds.add_locked_funds(current_epoch, vesting_sum, self.proving_period_start, spec);
-        self.locked_funds += vesting_sum;
 
-        // save the updated vesting table state
-        self.save_vesting_funds(store, &vesting_funds)?;
+        // Finally, record the new locked-funds total.
+        self.locked_funds += vesting_sum;
 
         Ok(amount_unlocked)
     }
 
     /// Draws from vesting table and unlocked funds to repay up to the fee debt.
-    /// Returns the amount unlocked from the vesting table and the amount taken from
-    /// current balance. If the fee debt exceeds the total amount available for repayment
-    /// the fee debt field is updated to track the remaining debt.  Otherwise it is set to zero.
+    /// Returns the amount to burn and the total amount unlocked from the vesting table (both vested
+    /// and unvested) If the fee debt exceeds the total amount available for repayment the fee debt
+    /// field is updated to track the remaining debt. Otherwise it is set to zero.
     pub fn repay_partial_debt_in_priority_order<BS: Blockstore>(
         &mut self,
         store: &BS,
         current_epoch: ChainEpoch,
         curr_balance: &TokenAmount,
     ) -> Result<
         (
-            TokenAmount, // from vesting
-            TokenAmount, // from balance
+            TokenAmount, // fee to burn
+            TokenAmount, // total unlocked
         ),
         anyhow::Error,
     > {
-        let unlocked_balance = self.get_unlocked_balance(curr_balance)?;
```

Comment:
continued paranoia: changes to this code are scary and I'd love to avoid them

Reply:
This code has some pretty thorough tests; I moved this down _because_ the tests failed after I started unlocking vested funds.

I also really don't want to get into a stance of "we don't touch this code". I'd rather make it easier to understand and harder to mess up. The new logic is, IMO, simpler to reason about:

1. Unlock the funds we need to pay the debt.
2. Pay off the debt up to the amount of unlocked funds we have.

Reply:
Yeah I am ok with it overall now that its done and reviewed.  Being forced to review this again was also valuable.